persistence/azure-table/configuration_code_astp_[3,).partial.md

```diff
@@ -67,3 +67,15 @@ In case installers are enabled, but there's a need to opt out from creating the
 
 snippet: EnableInstallersConfigurationOptingOutFromTableCreation
 
+### Partitioning
+
+When storing sagas, saga IDs are by default deterministically derived from the saga data:  the correlation property name and the correlation property value.
+This deterministic ID is used as the partition key. The partition key is the key to achieving transactionality.
+
+There might be scenario's in which it's desirable to set a custom partition key:
+- When upgrading to the Azure Table Persistence, it's possible to make use of the added transactional features by setting the partition key for existing sagas that may have in-flight messages
+- To enable business operations to participate in the transaction before the saga has been created, the saga ID can be precomputed and set as the partition key
+
+To set a custom partition key, the `IProvidePartitionKeyFromSagaId` interface implementation can be injected into behaviors in the logical pipeline stage.
+
+snippet: BehaviorUsingIProvidePartitionKeyFromSagaId
```

Comment:
```suggestion

### Partitioning and compatibility helpers

In order to atomically store data within the same transaction all data written as part of the same message handling pipeline needs to share the same partition key. When storing sagas, saga IDs are by default deterministically derived from the saga data: the correlation property name and the correlation property value. So for conversations involving a saga as part of the message handling pipeline this determinstically derived ID might be a perfect fit as a partition key. Unfortunately the saga ID assignment happens quite late during the message handling and is not exposed to the user code. This makes it difficult to:

- Support transactional processing for existing sagas that where stored with the previous version of the persister and may have in-flight messages -or-
- Expose the deterministically created saga ID early enough to enable business operations to participate in the transaction before the saga has been created

To support the above scenarios the `IProvidePartitionKeyFromSagaId` interface implementation can be injected into behaviors in the logical pipeline stage.

snippet: BehaviorUsingIProvidePartitionKeyFromSagaId

The component will make sure to

- Derive the table to write the business data and the saga data to by looking for the necessary `TableInformation` or falling back to the default table
- Extract the saga id header if present and populate that header value as the partition key for previously created sagas
- When no header information is available and the compatibility mode is enabled do the necessary secondary index lookups and depending on the configuration even fallback to table scanning  and populate the found saga ID as the partition key. -or-
- When no header information is available and the compatibility mode is disabled precompute the deterministc saga ID based on the provided correlation propert information and populate the precomputed value as the partition key.
```

Reply:
@lailabougria @SeanFeldman what do you think about this one? Might need to grammar tweaks though

Reply:
I incorporated your suggestion and made some grammar/wording changes @danielmarbach